### ISSUE TITLE
Failing unittest

### ISSUE THREAD

**Issue by stweil** [tesseract-ocr/tesseract#3215]
These tests fail in latest git master with `-fsanitize=address,undefined`:

    ligature_table_test.log
    mastertrainer_test.log
    pango_font_info_test.log
    stringrenderer_test.log

**Comment by egorpugin** [comment]
Did it pass before?
We may broke some vector freeing functions (GenericVector<T*> templates).
I tried to not touch them for some time. GenVec<T> is simpler to replace.

Anyways considering that there are no much smart pointers on the codebase, it is not very surprising.

**Comment by stweil** [comment]
Yes, all tests passed. `pango_font_info_test` for example fails since commit db43bb43dcb673167014f4203468a6553527e668.

**Comment by egorpugin** [comment]
> ligature_table_test.log
mastertrainer_test.log
pango_font_info_test.log
stringrenderer_test.log

Attach these logs.

**Comment by stweil** [comment]
`ligature_table-test` shows errors like this one:

    Running main() from ../../../../unittest/../googletest/googletest/src/gtest_main.cc
    [==========] Running 7 tests from 1 test suite.
    [----------] Global test environment set-up.
    [----------] 7 tests from LigatureTableTest
    [ RUN      ] LigatureTableTest.DoesFillLigatureTables
    unknown file: Failure
    C++ exception with description "locale::facet::_S_create_c_locale name not valid" thrown in SetUp().
    [  FAILED  ] LigatureTableTest.DoesFillLigatureTables (1 ms)
    [ RUN      ] LigatureTableTest.DoesAddLigatures
    unknown file: Failure
    C++ exception with description "locale::facet::_S_create_c_locale name not valid" thrown in SetUp().
    [  FAILED  ] LigatureTableTest.DoesAddLigatures (0 ms)
    [...]

`pango_font_info_test` and `stringrenderer_test` also have the same kind of problem.

**Comment by egorpugin** [comment]
Much better.
1.
Why all tests run?
```
    static std::locale system_locale("");
    std::locale::global(system_locale);
```
I think this does nothing and can be removed from all tests.
Try it locally to check if `ligature_table_test` passes without it.

2.
Just a leak on raw pointer (`mastertrainer.cpp:147:20`). We need to replace it with smart ptr. As well as other many places.

**Comment by stweil** [comment]
1. Setting the locale is part of the tests. See my comment https://github.com/tesseract-ocr/tesseract/commit/a52237799338580c761b13ccc2bfd8b8ffaf4edb#commitcomment-45833215 for more information.
2. The three other tests are fixed now, so fixing `mastertrainer_test` would close this issue.

**Comment by egorpugin** [comment]
1. Ok, good to know.

**Comment by stweil** [comment]
The last failing `mastertrainer_test` was fixed with pull request #3256, so this issue can now be closed.
